core/src/main/scala/com/salesforce/op/stages/impl/insights/RecordInsightsLOCO.scala

```diff
@@ -86,26 +124,102 @@ class RecordInsightsLOCO[T <: Model[T]]
 
     val k = $(topK)
     var i = 0
-    while (i < filledSize) {
-      val (oldInd, oldVal) = featureArray(i)
-      featureArray.update(i, (oldInd, 0))
-      val score = modelApply(labelDummy, OPVector(Vectors.sparse(featureSize, featureArray))).score
-      val diffs = baseScore.zip(score).map{ case (b, s) => b - s }
-      val max = diffs.maxBy(math.abs)
-      maxHeap.enqueue((i, max, diffs))
-      if (i >= k) maxHeap.dequeue()
-      featureArray.update(i, (oldInd, oldVal))
-      i += 1
+    val top = $(topKStrategy) match {
+      case s if s == TopKStrategy.Abs.entryName || problemType == ProblemType.MultiClassification => {
+        val absoluteMaxHeap = PriorityQueue.empty(AbsScore)
+        var count = 0
+        while (i < filledSize) {
+          val (oldInd, oldVal) = featureArray(i)
+          featureArray.update(i, (oldInd, 0))
+          val score = modelApply(labelDummy, OPVector(Vectors.sparse(featureSize, featureArray))).score
+          val diffs = baseScore.zip(score).map { case (b, s) => b - s }
+          val max = diffs.maxBy(math.abs)
+          if (max != 0) { // Not keeping LOCOs with value 0.0
+            absoluteMaxHeap.enqueue((i, max, diffs))
+            count += 1
+            if (count > k) absoluteMaxHeap.dequeue()
+          }
+          featureArray.update(i, (oldInd, oldVal))
+          i += 1
+        }
+
+        val topAbs = absoluteMaxHeap.dequeueAll
+
+
+        topAbs.sortBy { case (_, v, _) => -math.abs(v) }
+
+      }
+      case s if s == TopKStrategy.PositiveNegative.entryName => {
```

Comment:
what if we always kept the positive and negative heap and then just check the strategy when we return the insights?

Reply:
It would be slower when we want to return the top by abs value. Is there an easy way to go from 2 Priority Queues to 1?

Reply:
It would be slightly slower but not by much if we assume that (or enforce that) k is small. To go to one q - just get all the answers and sort them. Again you are right that it is slower but if we limit k to ~100 it is a pretty small difference.

Reply:
Or you can get the first of each que and then basically keep popping off the last q that you took from for comparison - then it is not much slower you just have a sightly larger memory footprint

Reply:
Another issue @leahmcguire. For Multiclassification, we return the highest absolute LOCO value, and for the other strategy we look at the winning category (baseScore). Getting the top Positive and negative for winning class doesn't guarantee the highest absolute value : what if this absolute value was from another class?

Reply:
Yeah - I think that the fact that we were just taking the highest absolute value was a mistake. It allows for things like baseScore = (0.1, 0.3, 0.6) change in score = (0.3, 0.1, 0.6) to be the difference we report, which seems wrong. Always choosing the change in the winning value will mean that the change we report always has to do with the score that one, even if it is not the largest change in score. So yes we would change strategy for multiclass in absolute value as well - but I think it is slightly more principled...WDYT?

Reply:
Now the question is should we apply the same logic for Binary Classification. 
So far the spikes were focused on the positive class (label 1.0) because we thought it was the the most interesting info to report.

However, recent spikes witness a strong relationship  between LOCO scores and prediction scores (cc @crupley) : Rows with the "highest scores" had on average its top absolute LOCOs being mostly positive. For "lowest scores", those locos are mostly negative.
One can also argue that `LOCO for class 0 = - LOCO for class 1`, hence getting top Positives and Negatives is sufficient. 
Maybe something we might need to add to the output is the base score.

Reply:
Hmm, that is interesting.  But wouldn't it be confusing to interpret for Binary if we choose the winning value because they are symetric? the scores will always some to 1 so a positive contribution to the prediction of 1 will be the same negative contribution to the prediction of 0. If we switch based on the prediction they will have to keep in mind the winning prediction to interpret the effect - this is something you need to do in multiclass but seems like an extra complication in binary.